### PR TITLE
[15.0][OU-ADD] website_sale: update XPath for "product_custom_text" template

### PR DESCRIPTION
**Pull request by pilarvargas-tecnativa**
cc @Tecnativa TT45029

@chienandalu @pedrobaeza please review

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Better to ping when you are sure that everything is correct. Please ping in that moment.

**Comment by pilarvargas-tecnativa** [comment]
> Better to ping when you are sure that everything is correct. Please ping in that moment.

I will always make sure that all changes have been uploaded for pinging. Thanks

File: openupgrade_scripts/scripts/website_sale/15.0.1.1/post-migration.py [hunk 1]
```diff
@@ -1,6 +1,24 @@
 from openupgradelib import openupgrade
 
 
+def update_xpath_for_product_custom_text(env):
+    """Update the XPath expression associated with the "product_custom_text"
+    template on the associated website. The XPath needs to be changed as part of the
+    migration from v14 to v15, and this custom template must also be updated to reflect
+    the new XPath."""
```
Comment (chienandalu) [inline openupgrade_scripts/scripts/website_sale/15.0.1.1/post-migration.py]:
```suggestion
    """Look for custom views website_sale.product_custom_text and update the content hook"""
```

File: openupgrade_scripts/scripts/website_sale/15.0.1.1/post-migration.py [hunk 2]
```diff
@@ -1,6 +1,24 @@
...
+        [
+            ("key", "=", "website_sale.product_custom_text"),
+            ("website_id", "!=", False),
+        ]
+    ):
+        new_arch = view.arch_db.replace(
+            "expr=\"//div[@id='product_details']\"",
+            "expr=\"//div[@id='o_product_terms_and_share']\"",
+        )
+        view.write({"arch_db": new_arch})
```
Comment (chienandalu) [inline openupgrade_scripts/scripts/website_sale/15.0.1.1/post-migration.py]:
```suggestion
        view.arch_db = view.arch_db.replace(
            "expr=\"//div[@id='product_details']\"",
            "expr=\"//div[@id='o_product_terms_and_share']\"",
        )
```

Comment (pilarvargas-tecnativa) [reply openupgrade_scripts/scripts/website_sale/15.0.1.1/post-migration.py]: done and tested locally, thanks!

**Review by chienandalu — APPROVED** [review]

File: openupgrade_scripts/scripts/website_sale/15.0.1.1/post-migration.py [hunk 3]
```diff
@@ -1,6 +1,20 @@
...
 
+def update_xpath_for_product_custom_text(env):
+    """Look for custom views website_sale.product_custom_text and update the content hook"""
+    for view in env["ir.ui.view"].search(
+        [
+            ("key", "=", "website_sale.product_custom_text"),
+            ("website_id", "!=", False),
+        ]
+    ):
+        view.arch_db = view.arch_db.replace(
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/website_sale/15.0.1.1/post-migration.py:12]: Is this working correctly in the website context?

Comment (chienandalu) [reply openupgrade_scripts/scripts/website_sale/15.0.1.1/post-migration.py:12]: In this case we don't need the context as the script treats every custom view (check the domain)

**Comment by pedrobaeza** [comment]
@pilarvargas-tecnativa this is not `[OU-ADD]`, but `[OU-FIX]` or `[OU-IMP]`. Please remember next time.

**Pull request merged**